src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/VisitNotificationEventService.kt

```diff
@@ -130,8 +130,12 @@ class VisitNotificationEventService(
   fun handlePrisonerReceivedNotification(notificationDto: PrisonerReceivedNotificationDto) {
     LOG.debug("PrisonerReceivedNotification notification received : {}", notificationDto)
     if (PrisonerReceivedReasonType.TRANSFERRED == notificationDto.reason) {
-      val affectedVisits = visitService.getFutureVisitsBy(notificationDto.prisonerNumber, notificationDto.prisonCode)
-      processVisitsWithNotifications(affectedVisits, PRISONER_RECEIVED_EVENT)
+      val prisonerDetails = prisonerService.getPrisoner(notificationDto.prisonerNumber)
+
+      val affectedVisits = prisonerDetails?.let { visitService.getFutureVisitsBy(it.prisonerId, prisonerDetails.lastPrisonCode) }
```

Comment:
I think a check for non null last Prison code is needed here. could add a test as well.

Reply:
implemented this thank you